dev-haskell/scientific/scientific-0.3.7.0-r2.ebuild

```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# ebuild generated by hackport 0.8.0.0.9999
+#hackport: flags: -integer-simple
+
+CABAL_HACKAGE_REVISION=5
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour test-suite"
+inherit haskell-cabal
+
+DESCRIPTION="Numbers represented using scientific notation"
+HOMEPAGE="https://github.com/basvandijk/scientific"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="amd64 ~arm64 ~ppc64 ~riscv ~x86"
+
+RDEPEND="
+	>=dev-haskell/hashable-1.2.7.0:=[profile?] <dev-haskell/hashable-1.5:=[profile?]
+	>=dev-haskell/integer-logarithms-1.0.3.1:=[profile?] <dev-haskell/integer-logarithms-1.1:=[profile?]
+	>=dev-haskell/primitive-0.7.1.0:=[profile?] <dev-haskell/primitive-0.8:=[profile?]
+	>=dev-lang/ghc-8.4.3:=
+"
+DEPEND="${RDEPEND}
+	>=dev-haskell/cabal-2.2.0.1
+	test? (
+		>=dev-haskell/quickcheck-2.14.2
+		>=dev-haskell/smallcheck-1.0
+		>=dev-haskell/tasty-1.4.0.1
+		>=dev-haskell/tasty-hunit-0.8
+		>=dev-haskell/tasty-quickcheck-0.8
+		>=dev-haskell/tasty-smallcheck-0.2
+	)
+"
+pkg_pretend() {
+	if use test; then
+		ewarn "The \"test\" USE flag for this package creates cycles within the"
+		ewarn "dependency graph. This may give you problems during 'haskell-updater' runs."
+		ewarn "It is recommended to leave it disabled unless explicitly testing the package."
+	fi
```

Comment:
For each of the packages where `USE=test` is masked, a warning is emitted if someone circumvents the mask and enables the flag.